### ISSUE TITLE
Refactor CommandLineResolver

### ISSUE THREAD

**Issue by IanMayo** [debrief/pepys-import#103]
This is a relatively low priority task, since it doesn't bring any new features. It's suited for when @BarisSari is held up on other tasks.
 
We've refactored the command line resolver, and adopted `prompt_toolkit` to enhance the command line support.

But, there are some repeating patterns in the code, where we do similar operations on a range of table types (such as fuzzy search for sensor, platform, datafile, etc).

_If possible_, it would be good to refactor these, to pull out the common code. This would be easier to maintain, and easier to maintain our test coverage. I expect we're starting with complete test coverage of the implementations - but we should be able to delete some, post-refactoring.

**Comment by robintw** [comment]
@BarisSari Has this been done? I remember that you extracted various common parts into functions like `resolve_reference`. Would you say this has been completed?

**Comment by BarisSari** [comment]
Yes, @robintw . I think this has been completed. I'm closing the issue.
